packages/library-base/src/commonMain/kotlin/io/realm/internal/SuspendableNotifier.kt

```diff
@@ -43,22 +43,30 @@ internal class SuspendableNotifier(
         }
     }
 
-    // FIXME Work-around for the global Realm changed listener not working.
     // Adding extra buffer capacity as we are otherwise never able to emit anything
     // see https://github.com/Kotlin/kotlinx.coroutines/blob/master/kotlinx-coroutines-core/common/src/flow/SharedFlow.kt#L78
-    private val _realmChanged = MutableSharedFlow<RealmReference>(
-        onBufferOverflow = BufferOverflow.SUSPEND,
+    private val _realmChanged = MutableSharedFlow<FrozenRealmReference>(
+        onBufferOverflow = BufferOverflow.DROP_OLDEST,
```

Comment:
I haven't traced the logic fully, but is it safe if events are dropped? I'm thinking both about memory leaks objects/Realm getting out of sync?

Reply:
The versions are now tracked on the `LiveRealm`s issuing the actual `FrozenRealmReference`s, instead of relying on the various frozen versions to actually realm the user facing realm. So we are now actually NOT leaking them because we are now also closing them even though the were not emitted to the user facing realm. Also, dropping the older versions ensures that the user facing realm will get the latest version when it is advanced instead of throwing all newer version away with a log entry indication.